src/nvim/main.c

```diff
@@ -145,6 +145,21 @@ void early_init(void)
 {
   handle_init();
 
+  // Use $NVIM and $NVIMRUNTIME as $VIM and $VIMRUNTIME
+  // or unset them to avoid accidentaly using the Vim runtime
+  const char *env_nvimruntime = os_getenv("NVIMRUNTIME");
+  if (env_nvimruntime) {
+    os_setenv("VIMRUNTIME", env_nvimruntime, true);
+  } else {
+    os_unsetenv("VIMRUNTIME");
+  }
+  const char *env_nvim = os_getenv("NVIM");
+  if (env_nvim) {
+    os_setenv("VIM", env_nvim, true);
```

Comment:
Let's leave this out for now, and instead just change the references (there's only ~6) in option.c, eval.c, ex_cmds.c, and env.c.

If the backwards compatibility thing turns out to be a nightmare, *then* we can consider setting `$VIM`. But it should be a last resort.